guide on how to work with product design (#4003)
PostHog/posthog.com: contents/handbook/engineering/product-design.md

---
title: Product Design, for Engineers
sidebar: Handbook
showTitle: true
---

It is up to you when to involve our product designers in your work.

## v0.1 or v2?

As the feature owner, you should make a choice if you're building a very basic first iteration of something, or if you're improving the experience.

If you're improving an existing feature that is popular, you are probably creating v2.

> If you're shipping an entirely new feature (ie SQL for PostHog), then you should be figuring out if anyone cares, which usually means creating a v0.1, and releasing it behind a feature flag to some friendly users (pro tip - make friends by being support hero).

Reviewer: ```suggestion
If you're shipping an entirely new feature (i.e. SQL for PostHog), then you should figure out if anyone cares, which usually means creating a v0.1, and releasing it behind a feature flag to some friendly users (pro tip: make friends by being support hero).
```

> Please _communicate_ to product design what your expectations are!

Reviewer 2: Can we clarify what are the requirements for the quality of the design in v0.1 and v2? 

Can we ever release something as a v2 without both an engineer's and a designer's ✅, especially if it involves UI changes?

I'd argue that if we want to keep a consistently high quality of product, a v2 or unflagged user facing feature should have the designer's blessing. Luckily they're easy to come by if you use standard components in familiar patterns, which are getting properly documented more and more.

Reviewer 3: Sometimes it might be worth looping into a designer at the very start to explore what it could be such as doing concepting work in deciding what the feature/product could be.

> Please _communicate_ what your expectation is to product design too here!

Reviewer: ```suggestion
Please _communicate_ to product design what your expectations are!
```

> ### T-shirt sizes

Reviewer 2: I don't get the reference to t-shirts, as I probably missed some meeting. Should I google this?

Reviewer 4: It's a relative measurement for estimating the complexity or amount of work required to solve a problem. Extra-small, small, medium, large, etc. It's really just a loose way of saying this is a small problem or a big problem.

Author: todo: i'll reword above to remove the assumption people know this term

> The more complex a feature to implement, the more likely it is that involving product design will make you faster.

Reviewer: ```suggestion
The more complex a feature is to implement, the more likely it is that involving product design will make you faster.
```

### Your design skill

We generally hire full stack engineers, but some people think more like designers than others. This is fine - you should play to your strengths.

The less strong you are at design, the more we'd encourage you to involve a product designer.

If you're unsure about your skill level, ask a product designer for direct feedback. [This is a book we'd recommend](https://en.wikipedia.org/wiki/The_Design_of_Everyday_Things) if you want to learn the mindset.

### If product design has capacity

> Sometimes product design may push back if they simply haven't got capacity. It's subjective when this may happen, and it'll usually be in cases where they feel they won't be as helpful based on the above.

Reviewer 2: This part makes it feel like design is optional. If so, we should clearly state that's the case. 

I'd still argue for clearly making it optional but suggested for v0.1 (that you release to beta users just to get a response), but necessary for a public unflagged and marketed v2. We can also agree that there will be exceptions or emergencies where we step on toes, and agree that we will not do so by default.

If we don't have enough product design capacity to get approvals in a timely manner, let's hire more 🤷.

---
title: Product Design, for Engineers
sidebar: Handbook
showTitle: true
---

It is up to you when to involve our product designers in your work. Because we hire generalists, there is no expectation that every project should start by running through design _first_. 

You should start by identifying the stage and goals of your project.

## v0.1 or v2?

As the feature owner, you should make a choice if you're building a very basic first iteration of something, or if you're improving the experience.

There are two paths for creating the first version of a product: v0.1 or MVP (even earlier).

### v0.1

If we're attempting to reach parity on a product or feature with other competitors in the space – and there's a clear path toward how a product should work or look – there's no need to loop in a designer. You should make your best judgement, while leveraging our [design system](https://storybook.posthog.net) to build your feature.

### MVP

> If you're shipping an entirely new feature (i.e. SQL for PostHog), then you should figure out if anyone cares, which usually means creating an MVP and releasing it behind a feature flag to some friendly users. (Pro tip: make friends by being [support hero](/handbook/engineering/support-hero).)

Author: ```suggestion
If you're shipping an entirely new feature (i.e. SQL for PostHog), then you should figure out if any users even care (!), which usually means creating an MVP and releasing it behind a feature flag to some friendly users. (Pro tip: make friends by being [support hero](/handbook/engineering/support-hero).)
```
